Makefile

```diff
@@ -3,6 +3,7 @@ SRCS_JS_FROM_TS =
 
 SRCS_TS += index.ts
 SRCS_TS += $(shell find lib -type f -name '*.ts')
+SRCS_TS += $(shell find test -type f -name '*.ts')
```

Comment:
I can do this since I already am somewhat familiar with it.  @ericlu88, it may be best to move the testing changes into a separate PR since this may take some time.  Is that cool with you?